core/src/main/java/org/springframework/security/access/expression/method/DefaultMethodSecurityExpressionHandler.java

```diff
@@ -286,16 +289,24 @@ public void setReturnObject(@Nullable Object returnObject, EvaluationContext ctx
 	 * If null or empty, then no default role prefix is used.
 	 * </p>
 	 * @param defaultRolePrefix the default prefix to add to roles. Default "ROLE_".
+	 * @deprecated Use
+	 * {@link #setAuthorizationManagerFactory(AuthorizationManagerFactory)} instead
 	 */
-	public void setDefaultRolePrefix(String defaultRolePrefix) {
-		this.defaultRolePrefix = defaultRolePrefix;
+	@Deprecated(since = "7.0")
+	public void setDefaultRolePrefix(@Nullable String defaultRolePrefix) {
+		if (defaultRolePrefix == null) {
+			defaultRolePrefix = "";
+		}
```

Comment:
This is another workaround for passivity, similar to stripping the role prefix. These expression handlers were allowing null or blank, but the new implementation only allows a blank prefix.